### PR TITLE
feat:[SSCA-1148]: Open API spec for CreateTicket

### PR DESCRIPTION
**Pull request by humanshu2002**
## Describe your changes
Created the open OpenApi spec for create ticket
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56997)

### CODE REVIEW

File: ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java [hunk 1]
```diff
@@ -7,11 +7,7 @@
 package io.harness.ssca.api;
 
 import io.harness.spec.server.ssca.v1.RemediationApi;
-import io.harness.spec.server.ssca.v1.model.ExcludeArtifactRequestBody;
-import io.harness.spec.server.ssca.v1.model.RemediationTrackerCreateRequestBody;
-import io.harness.spec.server.ssca.v1.model.RemediationTrackerCreateResponseBody;
-import io.harness.spec.server.ssca.v1.model.RemediationTrackersOverallSummaryResponseBody;
-import io.harness.spec.server.ssca.v1.model.SaveResponse;
+import io.harness.spec.server.ssca.v1.model.*;
```
Comment (mankrit-singh) [inline ssca-manager/service/src/main/java/io/harness/ssca/api/RemediationTrackerApiImpl.java:10]: let's not add * imports

File: ssca-manager/contracts/openapi/v1/openapi.yaml [hunk 2]
```diff
@@ -2625,6 +2663,79 @@ components:
...
+            idName:
+              - value1
+              - value2
+              - value3
+          issueType: Bug
+          priority: High
+          projectKey: ABC
+          title: A new ticket
+      properties:
+        customFields:
```
Comment (mankrit-singh) [inline ssca-manager/contracts/openapi/v1/openapi.yaml:2689]: all of these should be in snake case
eg. custom_fields etc.

File: ssca-manager/contracts/openapi/v1/openapi.yaml [hunk 3]
```diff
@@ -962,20 +962,58 @@ paths:
...
+        id: vw1a3b5us3lxi
       description: Exclude Artifact From Remediation Tracker.
       security:
-        - x-api-key: [ ]
+        - x-api-key: []
       parameters:
         - $ref: '#/components/parameters/AccountHeader'
       requestBody:
         $ref: '#/components/requestBodies/ExcludeArtifactRequestBody'
+  '/v1/orgs/{org}/projects/{project}/remediation/{remediation}/create-ticket':
```
Comment (mankrit-singh) [inline ssca-manager/contracts/openapi/v1/openapi.yaml:973]: please make the path - /v1/orgs/{org}/projects/{project}/remediations/{remediation}/create-ticket

File: ssca-manager/contracts/openapi/v1/openapi.yaml [hunk 4]
```diff
@@ -962,20 +962,58 @@ paths:
...
+    post:
+      summary: Create Ticket
+      operationId: createTicket
+      responses:
+        '200':
```
Comment (mankrit-singh) [inline ssca-manager/contracts/openapi/v1/openapi.yaml:978]: this response should be 201

**Comment by humanshu2002** [comment]
trigger uts

**Comment by humanshu2002** [comment]
trigger uts

**Comment by humanshu2002** [comment]
trigger uts

**Review by mankrit-singh — APPROVED** [review]

**Comment by humanshu2002** [comment]
trigger ut5,ut6,ut7,ut8,ut9

**Comment by humanshu2002** [comment]
trigger ut6,ut8

**Comment by humanshu2002** [comment]
trigger ut5,ut6,ut7,ut8,ut9

**Comment by humanshu2002** [comment]
trigger ut0,ut3

**Pull request merged**